integration_test/tests/power_bi_tests.rs

```diff
@@ -244,4 +246,149 @@ mod integration {
             );
         }
     }
+
+    /// Test PowerBi data retrieval flow (setup and connection flows are prerequisites)
+    /// data retreival flow is:
+    ///     - SQLAllocHandle(SQL_HANDLE_STMT)
+    ///     - SQLGetInfoW(SQL_DRIVER_ODBC_VER)
+    ///     - SQLGetInfoW(SQL_DRIVER_NAME)
+    ///     - SQLExecDirectW({NullTerminatedQuery},SQL_NTS)
+    ///     - SQLGetFunctions(SQL_API_SQLFETCHSCROLL)
+    ///     - SQLGetInfoW(SQL_GETDATA_EXTENSIONS)
+    ///     - SQLNumResultCols()
+    ///     - For columns 1 to {numCols}
+    ///         - SQLColAttributeW(SQL_DESC_CONCISE_TYPE)
+    ///         - SQLColAttributeW(SQL_DESC_UNSIGNED)
+    ///         - SQLColAttributeW(SQL_COLUMN_NAME)
+    ///         - SQLColAttributeW(SQL_COLUMN_NULLABLE)
+    ///         - SQLColAttributeW(SQL_DESC_TYPE_NAME)
+    ///         - SQLColAttributeW(SQL_COLUMN_LENGTH)
+    ///         - SQLColAttributeW(SQL_COLUMN_SCALE)
+    ///     - For X rows or until SQLFetch returns SQL_NO_DATA
+    ///         - SQLFetch()
+    ///         - For columns 1 to {numCols}
+    ///             - SQLGetData({colIndex}, {defaultCtoSqlType})
+    ///     - SQLMoreResults()
+    ///     - SQLFreeHandle(SQL_HANDLE_STMT)
+    #[test]
+    fn test_data_retrieval() {
+        let env_handle: HEnv = setup();
+        let (conn_handle, _, _, _) = power_bi_connect(env_handle);
+        let mut stmt: Handle = null_mut();
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLAllocHandle(
+                    HandleType::Stmt,
+                    conn_handle as *mut _,
+                    &mut stmt as *mut Handle
+                )
+            );
+
+            // SQL_DRIVER_ODBC_VER and SQL_DRIVER_NAME are not available through odbc_sys
+            /*
+            SQLGetInfoW(SQL_DRIVER_ODBC_VER)
+            SQLGetInfoW(SQL_DRIVER_NAME)
+            */
+
+            let mut query: Vec<u16> = "select * from example".encode_utf16().collect();
+            query.push(0);
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLExecDirectW(stmt as HStmt, query.as_ptr(), SQL_NTS as i32)
```

Comment:
If you don't mind, can you update the asserts for the API calls  slightly so that if the result is not SUCCESS, it prints out the diagnostic? You can look at power_bi_connect and SQLDriverConnectW

Reply:
👍, would have been helpful for ^ also 😄